docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -3,8 +3,293 @@ title: Excel custom functions tutorial
+
+## Create a custom function that requests data from the web
+
+What if you needed a function that could request the price of a stock from an API and display the result in the cell of a worksheet? Custom functions are designed so that you can easily request data from the web asynchronously.
+
+Complete the following steps to create a custom function named `stockPrice` that accepts a stock ticker symbol (e.g., **MSFT**) and returns the price of that stock. This custom function uses the IEX Trading API, which is free and does not require authentication.
+
+1. In the **stock-ticker** project that the Yeoman generator created, find the file **src/customfunctions.js** and open it in your code editor.
+
+2. In **customfunctions.js**, locate the `increment` function and add the following code immediately after that function.
+
+    ```js
+    function stockPrice(ticker) {
+        var url = "https://api.iextrading.com/1.0/stock/" + ticker + "/price";
+        return fetch(url)
+            .then(function(response) {
+                return response.text();
+            })
+            .then(function(text) {
+                return parseFloat(text);
+            });
+
+        // Note: in case of an error, the returned rejected Promise
+        //    will be bubbled up to Excel to indicate an error.
+    }
+
+3. In **customfunctions.js**, locate the line`CustomFunctionMappings.INCREMENT = increment;`, add the following line of code immediately after that line, and save the file.
+
+    ```js
+    CustomFunctionMappings.STOCKPRICE = stockPrice;
+    ```
+
+4. Before Excel can make this new function available, you must specify metadata to describe the function to Excel. Open the **config/customfunctions.json** file. Add the following JSON object to the 'functions' array and save the file.
+
+    This JSON describes the `stockPrice` function.
+
+    ```JSON
+    {
+        "id": "STOCKPRICE",
+        "name": "STOCKPRICE",
+        "description": "Fetches current stock price",
+        "helpUrl": "http://www.contoso.com/help",
+        "result": {
+            "type": "number",
+            "dimensionality": "scalar"
+        },  
+        "parameters": [
+            {
+                "name": "ticker",
+                "description": "stock ticker name",
+                "type": "string",
+                "dimensionality": "scalar"
+            }
+        ]
+    }
+    ```
+
+5. You must re-register the add-in in Excel in order for the new function to be available to end-users. Complete the following steps for the platform that you're using in this tutorial.
+
+    * If you're using Excel for Windows:
+
+        1. Close Excel and then reopen Excel.
+
+        2. In Excel, choose the **Insert** tab and then choose the down-arrow located to the right of **My Add-ins**.
+            ![Insert ribbon in Excel for Windows with the My Add-ins arrow highlighted](../images/excel-cf-register-add-in-1b.png)
```

Comment:
FYI, "1b" is highlighted in red in the GitHub UI. Please double check this works as intended.

Reply:
I noticed that too; image renders fine.